setup-slave.sh

```diff
@@ -14,7 +14,7 @@ source ec2-variables.sh
 
 # Set hostname based on EC2 private DNS name, so that it is set correctly
 # even if the instance is restarted with a different private DNS name
-PRIVATE_DNS=`wget -q -O - http://instance-data.ec2.internal/latest/meta-data/local-hostname`
+PRIVATE_DNS=`wget -q -O - http://169.254.169.254/latest/meta-data/local-hostname`
```

Comment:
I thought about creation of "module" and putting it in the list of modules in spark-ec2.py before "spark" module. 

It might be better to do it your way.